### PR TITLE
Add support for Face ID/Touch ID to app lock.

### PR DESCRIPTION
**Pull request by pixlwave**
Adds biometric unlock support to the AppLock feature. Should be testable now, although there are still some edge cases I need to handle in follow-up PRs.

Enabling with PIN code already setup (for illustration, the grace period is turned off):

https://github.com/vector-im/element-x-ios/assets/6060466/4c1f7ccd-b18d-4254-b1ef-e1f0db82beaf

### CODE REVIEW

File: ElementX/Resources/Localizations/en.lproj/Untranslated.strings [hunk 1]
```diff
@@ -4,6 +4,8 @@
 /* Used for testing */
 "untranslated" = "Untranslated";
 
+"screen_app_lock_biometric_prompt_reason" = "Authentication is needed to access your app";
```
Comment (pixlwave) [inline ElementX/Resources/Localizations/en.lproj/Untranslated.strings:7]: Add to Localazy before merging.

File: ElementX/Sources/FlowCoordinators/AppLockFlowCoordinator.swift [hunk 2]
```diff
@@ -32,6 +32,8 @@ class AppLockFlowCoordinator: CoordinatorProtocol {
     let userIndicatorController: UserIndicatorController
     let navigationCoordinator: NavigationRootCoordinator
     
+    /// A task used to await biometric unlock before showing the PIN screen.
+    private var unlockTask: Task<Void, Never>?
```
Comment (alfogrillo) [inline ElementX/Sources/FlowCoordinators/AppLockFlowCoordinator.swift:36]: Maybe an occasion for using `@CancellableTask`?

File: ElementX/Sources/FlowCoordinators/AppLockFlowCoordinator.swift [hunk 3]
```diff
@@ -71,6 +76,8 @@ class AppLockFlowCoordinator: CoordinatorProtocol {
     // MARK: - App unlock
     
     private func applicationDidEnterBackground() {
+        unlockTask?.cancel()
```
Comment (alfogrillo) [inline ElementX/Sources/FlowCoordinators/AppLockFlowCoordinator.swift:79]:
```suggestion
        unlockTask = nil
```
If you decide to use `@CancellableTask`

File: ElementX/Sources/FlowCoordinators/AppLockSetupFlowCoordinator.swift [hunk 4]
```diff
@@ -140,6 +140,8 @@ class AppLockSetupFlowCoordinator: FlowCoordinatorProtocol {
                 showBiometricsPrompt()
             case (.createPIN, .settings):
                 navigationStackCoordinator.setSheetCoordinator(nil)
+                #warning("Above is fine for change pin, but not create PIN with no biometrics.")
+            // showSettings()
```
Comment (alfogrillo) [inline ElementX/Sources/FlowCoordinators/AppLockSetupFlowCoordinator.swift:144]: Do you need some rework here?

**Review by alfogrillo — APPROVED** [review]
LGTM.
Great work! 👍🏻

File: ElementX/Sources/FlowCoordinators/AppLockFlowCoordinator.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/FlowCoordinators/AppLockFlowCoordinator.swift:36]: Oh yep, I forgot about that 😄, nice!

File: ElementX/Sources/FlowCoordinators/AppLockSetupFlowCoordinator.swift (hunk 4 above)
Comment (pixlwave) [reply ElementX/Sources/FlowCoordinators/AppLockSetupFlowCoordinator.swift:144]: Ah yeah, I made of note of this for a future PR, will remove the warning.

**Pull request merged**